### PR TITLE
Features/bagpipechanges

### PR DESCRIPTION
**Pull request by oyvindam**
Updated input with proper numerical with decimals, added onblur and onchanged. 
Updated stories for Input. (Proposed as new template for all components.)
Added component sorting in storybook.
Removed some crappy lint rules.
Added new numeric component
Introduced baseprops for all (most) components. 
Placeholder on select.

### CODE REVIEW

File: src/components/Input/Input.tsx [hunk 1]
```diff
@@ -39,36 +47,74 @@ const Input: FC<
...
-          {unit && <div className={`${style.unit}`}>unit</div>}
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const _previousValue = currentValue;
+      const _inputValue = event.target.value;
+      if (type === "number") {
+        // builtin type=number is useless, we need to validate the input ourselves
+        const decimalsPart = decimals ? `([.,]?[0-9]{0,${decimals}})` : '';
+        const regex = new RegExp(`^[0-9]*${decimalsPart}$`); // No global on this, it kills the regex
+        const valOk = regex.test(_inputValue);
+        console.log(valOk, _previousValue, _inputValue);
```
Comment (nwinger) [inline src/components/Input/Input.tsx:63]: nit: Remove. 👀

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -6,13 +6,19 @@ import "../../styles/tokens.css";
 import { sanitizeForId } from "../../utils";
 import { LabelPosition } from "../../types";
 import { Icon } from "../icon";
+import React from "react";
 
+/** Props for the Input */
 export type InputProps = {
   id: string;
   name: string;
+  type: 'text' | 'number';
```
Comment (nwinger) [inline src/components/Input/Input.tsx:15]: Notice how we extend the prop type defintion of the component, and spread {...props}. The idea was that we don't have to expose functions for already native functionality.

File: .storybook/preview.ts [hunk 3]
```diff
@@ -13,10 +13,15 @@ const preview: Preview = {
       defaultTheme: "light",
     }),
   ],
+
   parameters: {
     backgrounds: {
       disable: true,
     },
+    options: {
```
Comment (nwinger) [inline .storybook/preview.ts:21]: 💯

**Review by nwinger — CHANGES_REQUESTED** [review]
Have a look at the prop types for the Input :)

File: src/components/Input/Input.tsx (hunk 2 above)
Comment (oyvindam) [reply src/components/Input/Input.tsx:15]: Right. Moving the numerical functionality to specific type Numeric and leaving the input in peace.

**Pull request merged**